### PR TITLE
Bugfix: assign experiment id

### LINKED ISSUE #996: Bug: experiment id not properly attributed when calling the load_experiment_by_name method
Steps to reproduce: 

```python
from qcodes.dataset.measurements import Measurement

# Start with a clean data base
db_location = qcodes.config["core"]["db_location"]
db = DataSet(db_location)
exp = new_experiment("test", "test1")
exp_loaded = load_experiment_by_name("test", "test1")

# The following will work
meas = SweepMeasurement(exp=exp_loaded)
with meas.run() as datasaver: 
    pass

# This time we will have an error
with meas.run() as datasaver: 
    pass
```
If the experiment was already there in the database, the first measurement will also fail.

### PR DESCRIPTION
**Pull request by sohailc**
Fixes #996

### CODE REVIEW

File: qcodes/dataset/experiment_container.py [hunk 1]
```diff
@@ -240,4 +240,5 @@ def load_experiment_by_name(name: str,
         raise ValueError(f"Many experiments matching your request found {_repr}")
     else:
         e.exp_id = rows[0]['exp_id']
+        e.id = rows[0]['exp_id']  # Fix for issue 996
```
Comment (WilliamHPNielsen) [inline qcodes/dataset/experiment_container.py]: I don't see a reason for this comment?

File: qcodes/tests/dataset/test_experiment_container.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        qc.dataset.experiment_container.debug_db = qc.config["core"]["db_debug"]
+        _c = connect(qc.config["core"]["db_location"], qc.config["core"]["db_debug"])
+        init_db(_c)
+        _c.close()
+        yield
+
+
+def test_run_loaded_experiment(empty_temp_db):
+    """
+    Test whether issue 996 as reported here:
```
Comment (WilliamHPNielsen) [inline qcodes/tests/dataset/test_experiment_container.py]: Again, I don't think we should duplicate git's work into the comments of the code.

File: qcodes/tests/dataset/test_experiment_container.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import pytest
+import tempfile
+import os
+
+import qcodes as qc
+from qcodes.dataset.data_set import DataSet
```
Comment (WilliamHPNielsen) [inline qcodes/tests/dataset/test_experiment_container.py]: Codacy claims that this import is unused.

**Comment by sohailc** [comment]
@WilliamHPNielsen I have processed your comments. Is the PR better now?

**Comment by WilliamHPNielsen** [comment]
@sohailc It surely is! 
💃

**Pull request merged**